engine/src/main/java/com/google/android/fhir/sync/Sync.kt

```diff
@@ -74,22 +90,24 @@ object Sync {
    *
    * @param periodicSyncConfiguration configuration to determine the sync frequency and retry
    *   mechanism
-   * @return a [Flow] of [SyncJobStatus]
+   * @return a [Flow] of [PeriodicSyncState]
    */
   @ExperimentalCoroutinesApi
   inline fun <reified W : FhirSyncWorker> periodicSync(
     context: Context,
     periodicSyncConfiguration: PeriodicSyncConfiguration,
-  ): Flow<SyncJobStatus> {
+  ): Flow<PeriodicSyncState> {
     val uniqueWorkName = "${W::class.java.name}-periodicSync"
     val flow = getWorkerInfo(context, uniqueWorkName)
+    val periodicWorkRequest =
+      createPeriodicWorkRequest(periodicSyncConfiguration, W::class.java, uniqueWorkName)
     WorkManager.getInstance(context)
       .enqueueUniquePeriodicWork(
         uniqueWorkName,
         ExistingPeriodicWorkPolicy.KEEP,
-        createPeriodicWorkRequest(periodicSyncConfiguration, W::class.java),
+        periodicWorkRequest,
       )
-    return flow
+    return combineSyncStateForPeriodicSync(context, uniqueWorkName, flow, periodicWorkRequest.id)
```

Comment:
same comment as above. 

```suggestion
    return combineSyncStateForPeriodicSync(context, uniqueWorkName, flow)
```

Reply:
done.